xarray/tests/test_interp.py

```diff
@@ -881,3 +881,23 @@ def test_interp1d_bounds_error():
 
     # default is to fill with nans, so this should pass
     da.interp(time=3.5)
+
+
+@requires_scipy
+@pytest.mark.parametrize(
+    "x, expected",
+    [
+        (2.5, True),
+        (np.array([2.5, 5]), True),
+        (("x", np.array([0, 0.5, 1, 2]), dict(unit="s")), False),
+    ],
+)
+def test_coord_attrs(x, expected):
+    base_attrs = dict(foo="bar")
+    ds = xr.Dataset(
+        data_vars=dict(a=2 * np.arange(5)),
+        coords={"x": ("x", np.arange(5), base_attrs)},
+    )
+
+    actual = ds.interp(x=x).x.attrs == base_attrs
+    assert expected == actual
```

Comment:
```suggestion
    actual = ds.interp(x=x)
    expected = ds
    assert_identical(expected, actual)
```
I think you can use `assert_identical` here, which compares attributes.

Reply:
Didn't work. An interpolated dataset is not identical to the original dataset. I think we'll stick to the old version.

Reply:
Ah I thought you interpolated at `ds.x` and not at `x`. I found it non-straightforward to understand - maybe?

```suggestion
    has_same_attrs = ds.interp(x=x).x.attrs == base_attrs
    assert expect_same_attrs == has_same_attrs
```
(I am also fine leaving it as - up to you if you want to change it (of course you'd need to change the name in parametrize` and the signature))

Reply:
I think you can solve this by passing the precomputed expected value as a parameter (which should be simple if you choose the input values accordingly) to the test function. The advantage would be that we get a better diff.

Reply:
If I understand correctly you want to use an attrs value instead of a boolean check? I thought about that and concluded that it will yield duplicating input values and therefore increasing the risk further for grammar errors. I think it's easier to see it as, if I interpolate with this x-value will I still have that same attrs value as before.

Reply:
no, I meant that you precompute the expected result. So for `2.5` you'd get the expected interpolation of `5` (I think) which you pass as `expected_values`:
```python
dim = () if isinstance(x, float) else "x"
expected = xr.Dataset({"a": (dim, expected_values)}, {"x": x})
actual = ds.interp(x=x)
assert_identical(actual, expected)
```

Reply:
@keewis, your `expected` ds doesn't include any attrs for the x coord now. So then you need to add a 3rd parameter, expected_x_values. So now we have 3 parameters that each will have slightly different formats increasing the complexity and losing focus on what this test is doing.

This exercise in pytest-cleverness is starting to lose its shine for me. You're welcome to suggest a working version but I don't want to put more effort into this as I think the current version does the job just fine.

Reply:
that's true. We would either have to pass the expected Dataset / DataArray as a single `expected` parameter or keep it as is. Again, the advantage of using `assert_identical` would be that the diff is easier to read and failures are easier to debug, but I agree that it might not be worth the additional complexity.